support-frontend/assets/pages/[countryGroupId]/checkout.tsx

```diff
@@ -346,13 +146,104 @@ const processPayment = (statusResponse: StatusResponse) => {
 			})
 				.then((response) => response.json())
 				.then((json) => {
-					processPayment(json as StatusResponse);
+					void processPayment(json as StatusResponse, geoId);
 				});
 		}, 1000);
 	}
 };
 
-export function Checkout() {
+/** QueryString - this is setup specifically for the checkout page */
+function isNumeric(str: string) {
+	return !isNaN(parseFloat(str));
+}
+const searchParams = new URLSearchParams(window.location.search);
+const searchParamsPrice = searchParams.get('price');
+const query = {
+	product: searchParams.get('product') ?? '',
+	ratePlan: searchParams.get('ratePlan') ?? '',
+	price:
+		searchParamsPrice && isNumeric(searchParamsPrice)
+			? parseFloat(searchParamsPrice)
+			: undefined,
+};
+
+type Props = {
+	geoId: GeoId;
+};
+function CheckoutComponent({ geoId }: Props) {
```

Comment:
The ordering of this file has changed - that's provably the most important piece to review.